ERCS/erc-7573.md

```diff
@@ -0,0 +1,102 @@
+by the method
+```solidity
+function transferAndDecrypt(bytes32 id, address to, string memory keyEncryptedSuccess, string memory keyEncryptedFailure) external;
+```
+
+The contract  on the asset chain, implementing `ILockingContract`, will lock the token until a key is presented
+releasing the token by calling
+```solidity
+function transferWithKey(bytes32 id, string memory key) external;
+```
+where `key` is the decrypted `keyEncryptedSuccess` or `keyEncryptedFailure`
+
+For the decryption, a stateless decryption oracle is attached to the chain running `IDecryptionContract`.
+
+## Motivation
+
+Within the domain of financial transactions and distributed ledger technology (DLT), the Hash-Linked Contract (HLC) concept has been recognized as valuable and has been thoroughly investigated.
+The concept may help to solve the challenge of delivery-versus-payment (DvP), especially in cases where the asset chain and payment system (which may be a chain, too) are separated. The proposed solutions are based on an API-based interaction mechanism which bridges the communication between a so-called Asset Chain and a corresponding Payment System or requires complex and problematic time-locks (\cite{BancaItalia}). We believe an even more lightweight interaction across both systems is possible, especially when the payment system is also based on a DLT infrastructure.
+
+## Specification
+
+### Methods
+
+#### Smart Contract on one chain (e.g. the asset chain)
+
+```solidity
+interface ILockingContract {
+    event TransferIncepted(address initiator, bytes32 id);
+    event TransferConfirmed(address confirmer, bytes32 id);
+    event TokenClaimed(bytes32 id, string key);
+    event TokenReclaimed(bytes32 id, string key);
+
+    function inceptTransfer(bytes32 id, int amount, address from, string memory keyEncryptedSeller) external;
+    function confirmTransfer(bytes32 id, int amount, address to, string memory keyEncryptedBuyer) external;
+    function transferWithKey(bytes32 id, string memory key) external;
+}
+```
+
+#### Smart Contract on the other chain (e.g. the payment chain)
+
+```solidity
+interface IDecryptionContract {
+    event TransferIncepted(address initiator, bytes32 id, int amount);
+    event TransferKeyRequested(bytes32 id, string encryptedKey);
+    event TransferKeyReleased(bytes32 id, bool success, string key);
+
+    function inceptTransfer(bytes32 id, int amount, address from, string memory keyEncryptedSuccess, string memory keyEncryptedFailure) external;
+    function transferAndDecrypt(bytes32 id, int amount, address to, string memory keyEncryptedSuccess, string memory keyEncryptedFailure) external;
+    function cancelAndDecrypt(bytes32 id, address from, string memory keyEncryptedSuccess, string memory keyEncryptedFailure) external;
+}
+```
+
+## Rationale
+
+The rationale is described in the following sequence diagram.
+
+### Sequence diagram of delivery versus payment
+
+![image info](../assets/eip-7573/doc/DvP-Seq-Diag.png)
```

Comment:
You can put this directly in the Specification section if you'd like. I think it suits that section more, since it illustrates the intended flow of the application.

Reply:
Done.